rules/experimental/rbe/README.md

```diff
@@ -0,0 +1,250 @@
+In this case, the WORKSPACE would, presumably, look like this:
+```
+other_re_exec_properties(
+    name = "exec_properties",
+)
+```
+
+And the targets in the BUILD files will be able to depend on targets from other repos that were
+written with RBE in mind, as the name of the repo defined in the WORKSPACE (exec_properties in this
+case) is the same. That is why the repo name exec_properties does *not* contain the word RBE.
+
+### Use case 4 - rbe_exec_properties with override_constants:
+
+Let's now assume that a particular repo, running with a particular RBE setup, wants to run
+all its remote execution actions without network access, possibly for the sake of identifying any
+network dependencies. This would be achieved as follows.
+
+In the WORKSPACE file, call
+```
+rbe_exec_properties(
+    name = "exec_properties",
+    override_constants = {
+        "NETWORK_ON": create_exec_properties_dict(docker_network = "off"),
+    },
+)
+```
+
+This would override the meaning of NETWORK_ON for this workspace only.
+For this override to work, we depend on targets marking their network dependecy by using NETWORK_ON
+that was loaded from the repo @exec_properties.
+
+### Use case 5 - custom execution properties
+
+In this scenario, let's assume that a target is best run remotely on a high memory GCE machine.
+Let's also assume that the RBE setup associated with the workspace where the target is defined has
+workers of type "n1-highmem-8".
+
+Setting exec_properties = {"gceMachineType" : "n1-highmem-8"} is problematic because it does not
+lend itself to another repo depending on this target in all cases. See
+[anti-patterns](#anti-patterns) below. Unlike the case of NETWORK_ON, rbe_exec_properties does not
+provide a standard HIGH_MEM_MACHINE execution property constant (although it might do so in the
+future).
+
+The recommended way for a repo (let's call this repo, repo A) to define this high-mem dependency is
+as follows:
+
+In the WORKSPACE file, call:
+```
+custom_exec_properties(
+    name = "repo_a_prefix_high_mem_machine_exec_property",
+    constants = {
+        "HIGH_MEM_MACHINE": create_exec_properties_dict(gce_machine_type = "n1-highmem-8"),
+    },
+)
+```
+
+And then in the BUILD file:
+```
+load("@repo_a_prefix_high_mem_machine_exec_property//:constants.bzl", "HIGH_MEM_MACHINE")
+foo_library(
+    name="my_lib",
+    ...
+    exec_properties = HIGH_MEM_MACHINE,
+)
+```
+
+A depending repo (repo B) must also call custom_exec_properties in their WORKSPACE and map
+HIGH_MEM_MACHINE.
+They may, for whatever reason, decide to map it to something else.
+```
+custom_exec_properties(
+    name = "repo_a_prefix_high_mem_machine_exec_property",
+    constants = {
+        "HIGH_MEM_MACHINE": ...,
+    },
+)
+```
+
+## Anti-patterns
+
+As alluded to in the use cases described above, there are some anti-patterns to avoid.
+
+### Anti-pattern 1 - Do not create repo names that are not properly prefixed.
```

Comment:
I would change the ordering and put this one last.
Most people would encounter 2 and 3. Only those who use custom_exec_properties would care about this one.

Reply:
Done.